### PR TITLE
tools: Move remaining monorepo JS dependencies to a project

### PR DESCRIPTION
**Pull request by anomiex**
#### Changes proposed in this Pull Request:

We did eslint in #20085. Let's rename that project to something more
generic and use it for semver, prettier, and sort-package-json too. And
our scripts in `tools/` (including the git hooks) and `.github/files/`
that had dependencies at the monorepo level as well.

#### Jetpack product discussion

None

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Is CI happy?
* Make a commit. Does the pre-commit hook run correctly?

### CODE REVIEW

File: tools/js-tools/git-hooks/pre-commit-hook.js [hunk 1]
```diff
@@ -119,10 +119,10 @@ function sortPackageJson( jsFiles ) {
 }
 
 const gitFiles = parseGitDiffToPathArray(
-	'git -c core.quotepath=off diff --cached --name-only --diff-filter=ACM'
+	'git -c core.quotepath=off diff --cached --name-only --diff-filter=ACMR'
 ).filter( Boolean );
 const dirtyFiles = parseGitDiffToPathArray(
-	'git -c core.quotepath=off diff --name-only --diff-filter=ACM'
```
Comment (jeherve) [inline tools/js-tools/git-hooks/pre-commit-hook.js:125]: For anyone who like me may be wondering, `R` stands for renamed. :) Seems to make sense here.

File: package.json [hunk 2]
```diff
@@ -74,24 +74,16 @@
...
-			"post-merge": "./tools/git-hooks/post-merge-checkout-hook.sh"
+			"pre-commit": "node tools/js-tools/git-hooks/pre-commit-hook.js",
+			"prepare-commit-msg": "node tools/js-tools/git-hooks/prepare-commit-msg.js",
+			"post-checkout": "./tools/js-tools/git-hooks/post-merge-checkout-hook.sh",
+			"post-merge": "./tools/js-tools/git-hooks/post-merge-checkout-hook.sh"
 		}
 	},
-	"dependencies": {
-		"chalk": "4.1.1"
-	},
```
Comment (jeherve) [inline package.json:85]: I notice we're moving this from a dependency to a dev dependency of the tools. Should we keep it around for `tools/check-renovate-ignore-list.js`?

Comment (anomiex) [reply package.json:85]: I'm not sure that it makes a difference, do we ever not install the JS dev dependencies? And if we don't, would that apply to some of the other deps here too?

Comment (jeherve) [reply package.json:85]: That's true, it's not really important for our monorepo tools. 👍

**Review by jeherve — APPROVED** [review]

**Comment by anomiex** [comment]
r228446-wpcom

**Pull request merged**